Made a python2.7.6 compatible example, using only core Python. Note that...

j-ruffer opened the pull request:
... the requests module would be more ideal, but it is not a core library, so I stuck to httplib.

This is a direct port of the HistoricalDataRequest.py example. I've tested this, though this should take some regression testing on other machines as well.

Commit:
Made a python2.7.6 compatible example, using only core Python. Note that the requests module would be more ideal, but it is not a core library, so I stuck to httplib.

ericvw commented on examples/python/python2/__main__.py:
```diff
@@ -0,0 +1,52 @@
+import json
+import sys
+import httplib
+import pprint
```
nit: It may be good to sort the `import`'s alphabetically.  Not necessary though.

ericvw commented:
First, thanks for helping to contribute a Python example.  A few things that would be great to see changed before we land this.
1. Please rename the file to something like `HistoricalDataRequest.py` so that we may have other named examples in this new directory.
2. It may be great to add some Python version checking and error reporting when the version isn't supported by the example to reduce the element of surprise for users
3. I can't tell from the GitHub diff viewer, but ensure that the file has the executable bit set.

ericvw commented:
Another thing, we tend follow these [commit message guidelines](https://github.com/blog/926-shiny-new-commit-styles).  If you could update your commit message accordingly, that would be awesome.  We relax the 50 character summary limit to something reasonably longer - use your own judgement.

j-ruffer commented:
Sure thing, I'll update according to your suggestions.

j-ruffer commented:
Import dependencies exist here, so alphabetical ording wouldn't be ideal. However, I can do a version check so we can get an early exit from the example. I'll update my commit message appropriately.

j-ruffer committed:
Python 2.7.6 HTTP API Historical Data Example

This example is compatible with Python 2.7.6, which requires use of the
low-level httplib for SSL connections, since the SSLContext class is not
available in the ssl module. The requests module would be more ideal here,
but is a third-party module, so was avoided in effort to stick to core python.

ericvw commented:
So this does not work in `2.7.9`:

``` bash
$ ./HistoricalDataRequest.py -s http-api.openbloomberg.com
http-api.openbloomberg.com
client.key
client.crt
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:581)
```

We can't in good faith have code examples that are security risks :).  It turns out that `httplib` in `<= 2.7.9` disable server verification, which lends itself to potential man-in-the-middle attacks.

If we can use `urllib2` or `urllib` to accomplish this properly, that would be preferred.

ericvw commented:
Just checking in on the status of this PR?  @j-ruffer, any updates?

j-ruffer commented:
@ericvw Still in progress. Would you like me to close until I have a proper example?

ericvw commented:
It is OK to keep open.  I just wanted to touch base and probably will again next Monday with status.

ericvw commented:
@j-ruffer, an updates?

ericvw commented:
Haven't heard back from original requester.

j-ruffer commented:
I'm cool with this. My bad!

Sent from Bloomberg Professional for Android

----- Original Message -----
From: Eric N. Vander Weele 
At: 08-Nov-2015 15:35:56

Closed #42.  
—
Reply to this email directly or view it on GitHub.
